### ISSUE TITLE
Images not parsed correctly if the path contains space

### ISSUE THREAD

**Issue by linusbergl** [greghendershott/markdown#60]
Images links are not parsed correctly if the path contains space

```
![hello](hel lo.jpg)
```
will not produce an image sexpr.

The commonmark spec says that the path to the image should conform to the same rules as a regular link. If you want a space in a regular link you should encapsulate it with <>, which would make the above example

```
![hello](<hel lo.jpg>)
```

which looks stupid, and also does not work with racket markdown.

Conforming to some kind of spec is of course nice, but wouldn't doing the expected thing (rendering the image anyway) also be nice? GitHub markdown does what I would expect (both examples work), pandoc as well.

Best regards
Linus

**Comment by greghendershott** [comment]
Thanks for the report.

It was already supposed to support the `<source with spaces>` syntax, but there was a bug in the grammar, which I just fixed.

The issue with spaces is that there's an optional title -- the general form for links and images is `[label](source "title")`. The idea behind requiring the `<>` is, I think, to parse `[label](<source with spaces> "some title")` more simply and reliably.

Admittedly, it's annoying if there is no optional title. Admittedly, even when a title is present, the parse could rely on the title being delimited with quotes, and the source being "everything before that". So, sure, I'll look at that, too. That would be fun to do, when I have time. But tweaking the parser risks creating bugs I don't have much spare time to fix now. As a result, if you're both very annoyed and very impatient you may want to submit a PR.

**Comment by linusbergl** [comment]
I'll look into it! I'm trying to grok the source code, but it goes way above my head as of right now. I have plans to extend the image syntax with a possibility for a CSS selector [caption](imgpath){css_class) for my static site generator (I want to float images left/right and generate thumbnails and all that jazz). Very much nonstandard and I understand if you are not interested.

Maybe fixing this issue could be a way to get acquainted with parsack and this markdown implementation. Maybe I have a PR before you look into it :)

**Comment by greghendershott** [comment]
To clarify now this will work: `![hello](<hel lo.jpg>)`. For example see [regression tests I just added](https://github.com/greghendershott/markdown/blob/master/markdown/test.rkt#L969-L987).

> Maybe fixing this issue could be a way to get acquainted with parsack and this markdown implementation. Maybe I have a PR before you look into it :)

That would be wonderful, and the odds are good that you would get to it before me. :)  I've been extremely busy with a new job lately. Although I've been able to keep up with maintaining my open source projects, just barely and not as quickly as I like.
